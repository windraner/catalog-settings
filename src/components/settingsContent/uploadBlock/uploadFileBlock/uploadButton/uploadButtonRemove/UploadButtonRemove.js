import React, { Component } from 'react';

import './uploadButtonRemove.css';

export default class UploadButtonRemove extends Component {

  clickHandler = () => {
    this.props.inputHandler('file', '');
  }

  render() {
    const { text } = this.props;

    return (
      <div className="catalog-settings-content__remove-upload-button">
        <span>{text}</span>
        <span
          onClick={this.clickHandler}
          className="catalog-settings-content__upload-button-remove-icon"
        />
      </div>
    );
  }
}
