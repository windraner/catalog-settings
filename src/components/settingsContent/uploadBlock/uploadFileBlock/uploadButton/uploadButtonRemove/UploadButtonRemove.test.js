import React from 'react';
import UploadButtonRemove from './UploadButtonRemove';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json'

configure({ adapter: new Adapter() })

describe('<UploadButtonRemove />', () => {
  it('Check for text prop', () => {
    const wrapper = shallow(<UploadButtonRemove inputHandler={function() {}} text="test" />);
    expect(wrapper.find('.catalog-settings-content__remove-upload-button').text()).toBe('test');
  });
  it('Check for click', () => {
    const onClickFunc = jest.fn();
    const wrapper = shallow(<UploadButtonRemove inputHandler={onClickFunc} text="" />);
    const button = wrapper.find('.catalog-settings-content__upload-button-remove-icon');
    button.simulate('click');
    expect(onClickFunc).toBeCalled();
    expect(onClickFunc.mock.calls).toEqual([["file", ""]]);
  });
  it('matches the snapshot', () => {
    const tree = shallow(<UploadButtonRemove
      inputHandler={function() {}}
      text=""
    />)
    expect(toJson(tree)).toMatchSnapshot()
  })
});
