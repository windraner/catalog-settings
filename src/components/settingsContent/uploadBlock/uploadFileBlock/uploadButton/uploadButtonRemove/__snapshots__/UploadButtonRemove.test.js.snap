// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UploadButtonRemove /> matches the snapshot 1`] = `
<div
  className="catalog-settings-content__remove-upload-button"
>
  <span />
  <span
    className="catalog-settings-content__upload-button-remove-icon"
    onClick={[Function]}
  />
</div>
`;
